include/fsm/fill_spill_merge.hpp

```diff
@@ -373,13 +375,15 @@ static void MoveWaterInDepHier(
     const auto lchild = this_dep.lchild;
     const auto rchild = this_dep.rchild;
 
+
     //Only if both children are full should their water make its way to this
     //parent
     if(lchild!=NO_VALUE
       && deps.at(lchild).water_vol==deps.at(lchild).dep_vol
       && deps.at(rchild).water_vol==deps.at(rchild).dep_vol
+      && deps.at(this_dep.dep_label).water_vol < FP_ERROR  
     )
-    this_dep.water_vol += deps.at(lchild).water_vol + deps.at(rchild).water_vol;
+    this_dep.water_vol += deps.at(lchild).dep_vol + deps.at(rchild).dep_vol;
```

Comment:
Why?

Reply:
Considering the conditional that water_vol == dep_vol in the line above, this change is not necessary.